sdk/database/helper/connutil/cloudsql.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package connutil
+
+import (
+	"fmt"
+
+	"cloud.google.com/go/cloudsqlconn"
+	"cloud.google.com/go/cloudsqlconn/postgres/pgxv4"
+)
+
+func (c *SQLConnectionProducer) getCloudSQLDriverName() (string, error) {
+	var driverName string
+	// using switch case for future extensibility
+	switch c.Type {
+	case dbTypePostgres:
+		driverName = cloudSQLPostgres
+	default:
+		return "", fmt.Errorf("unsupported DB type for cloud IAM: %s", c.Type)
+	}
+
+	return driverName, nil
+}
+
+func (c *SQLConnectionProducer) registerDrivers(driverName string, credentials string) (func() error, error) {
+	typ, err := c.getCloudSQLDriverName()
```

Comment:
nit: Wondering if `c.getCloudSQLDriverType()` would be more appropriate for a name? It confused me a bit when I first read it thinking that its output would be used for the driver name at registration.

Reply:
Updated!